### DISCUSSION TITLE
Is there a way to specify virtual machine region?

### DISCUSSION THREAD

**Discussion by dphang** [community/community#25994] [Actions]
<p>Does anyone know if there is a way to specify the shared runner’s virtual machine region used for Actions jobs? I did see that they are hosted in Azure regions: <a href="https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#ip-addresses-of-github-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#ip-addresses-of-github-hosted-runners</a></p>
<p>For example, is it possible to request an Ubuntu machine from <code>westus2</code>? The reason I need it is because I’m running some k6 API load tests (API is hosted in California), it would be nice to always have a machine from the same region so I can have consistent network latencies.</p>
<p>The alternatives that we considered:</p>
<ol>
<li>Use k6 Cloud. It supports running from different load zones, though it is a bit pricey at $49/month (we are a small startup, so we would like to save where possible).</li>
<li>Use a self-hosted runner. For example, we can create an AWS VM in a specific region and always run this specific workflow on this VM.</li>
<li>Trigger load test on AWS (EC2, Fargate, etc.) from deployment workflow and then once the load test completes, use GitHub API to continue the deployment.</li>
</ol>
<p>Anyone have other recommendations? <span class="hashtag">#2</span> seems to be the easiest option for now, as we would like to run it as part of the Actions deployment workflow.</p>
<p>Thanks in advance,<br>
Daniel</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I am afraid that I have to say there is not a way to specify region for GitHub hosted runner.  I would recommend you submit a feature request in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a>.</p>

**Reply by kohtala** [reply]
There is a related feature request #140310.

**Comment by nikimanoledaki** [comment]
+1 this feature would be highly valuable!
